src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -51,6 +52,32 @@ static void addConvertingActions(Pipe & pipe, const Block & header)
     });
 }
 
+static void enforceSorting(QueryProcessingStage::Enum stage, DataStream & output_stream, Context & context, SortDescription output_sort_description)
+{
+    if (stage != QueryProcessingStage::WithMergeableState)
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "Cannot enforce sorting for ReadFromRemote step up to stage {}",
+            QueryProcessingStage::toString(stage));
+
+    context.setSetting("enable_memory_bound_merging_of_aggregation_results", true);
+
+    output_stream.sort_description = std::move(output_sort_description);
+    output_stream.sort_scope = DataStream::SortScope::Stream;
+}
+
+static void enforceAggregationInOrder(QueryProcessingStage::Enum stage, Context & context)
+{
+    if (stage != QueryProcessingStage::WithMergeableState)
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "Cannot enforce aggregation in order for ReadFromRemote step up to stage {}",
+            QueryProcessingStage::toString(stage));
+
```

Comment:
```suggestion

context.setSetting("enable_memory_bound_merging_of_aggregation_results", true);
```

Reply:
I would rather separate enforceAggregationInOrder and enforceSorting calls.